contracts/src/v0.8/keystone/KeystoneForwarder.sol

```diff
@@ -97,29 +103,41 @@ contract KeystoneForwarder is IForwarder, ConfirmedOwner, TypeAndVersionInterfac
   uint256 internal constant FORWARDER_METADATA_LENGTH = 45;
   uint256 internal constant SIGNATURE_LENGTH = 65;
 
-  function setConfig(uint32 donId, uint8 f, address[] calldata signers) external onlyOwner {
+  function setConfig(uint32 donId, uint32 configVersion, uint8 f, address[] calldata signers) external onlyOwner {
     if (f == 0) revert FaultToleranceMustBePositive();
     if (signers.length > MAX_ORACLES) revert ExcessSigners(signers.length, MAX_ORACLES);
     if (signers.length <= 3 * f) revert InsufficientSigners(signers.length, 3 * f + 1);
 
-    // TODO: how does setConfig handle expiration? e.g. if the signer set changes
+    bytes32 configId = keccak256(abi.encode(donId, configVersion));
 
     // remove any old signer addresses
-    for (uint256 i; i < s_configs[donId].signers.length; ++i) {
-      address signer = s_configs[donId].signers[i];
-      delete s_configs[donId]._positions[signer];
+    for (uint256 i; i < s_configs[configId].signers.length; ++i) {
+      address signer = s_configs[configId].signers[i];
+      delete s_configs[configId]._positions[signer];
```

Comment:
This is no longer needed because the config ID will always increment, right?

Reply:
We still want to clear or there could be invalid values in the set if `setConfig` is called more than once for a config version.

Reply:
> if setConfig is called more than once for a config version.

When would this happen? Is it if we make a mistake and set the wrong values?